clientSecurity/src/main/java/org/apache/geode/examples/clientSecurity/ExampleAuthInit.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.examples.clientSecurity;
+
+import java.util.Properties;
+
+import org.apache.geode.LogWriter;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.security.AuthInitialize;
+import org.apache.geode.security.AuthenticationFailedException;
+
+public class ExampleAuthInit implements AuthInitialize {
+
+  private LogWriter logger;
+
+  static final String USER_NAME = "security-username";
+  static final String PASSWORD = "security-password";
+
+  public static AuthInitialize create() {
+    return new ExampleAuthInit();
+  }
+
+
+  public void init(LogWriter systemLogger, LogWriter securityLogger)
+      throws AuthenticationFailedException {
+    this.logger = securityLogger;
+  }
+
+  public Properties getCredentials(Properties securityProps, DistributedMember server,
+      boolean isPeer) throws AuthenticationFailedException {
+    Properties credentials = new Properties();
+    String userName = securityProps.getProperty(USER_NAME);
+    if (userName == null) {
```

Comment:
Add a comment here saying something like: "the implementer is supposed to use their existing infrastructure (ldap or anything else) to obtain the credentials and put them in the properties."
For the purpose of this example, I think it's ok if we relied on static final fields for credentials.